libraries/AP_GyroFFT/AP_GyroFFT.cpp

```diff
@@ -191,15 +215,18 @@ void AP_GyroFFT::init(uint32_t target_looptime_us)
     // determine the FFT sample rate based on the gyro rate, loop rate and configuration
     if (_sample_mode == 0) {
         _fft_sampling_rate_hz = _ins->get_raw_gyro_rate_hz();
+        for (uint8_t axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
+            _gyro_data[axis] = &_ins->get_raw_gyro_window(axis);
```

Comment:
the type handling here is very hard to follow. This seems to be taking a pointer to a GyroWindow and putting it into a pointer to a SampleWindow

Reply:
Fixed. This was a way of making the code simpler - but clearly makes it hard to understand! I have removed it.